### DISCUSSION TITLE
Nested key claim name

### DISCUSSION THREAD

**Discussion by meysam81** [Kong/kong#7277] [Ideas and feature requests]
I want to know how one can use nested jwt payload in kong?
imagine I have the following payload:
```
{
    'level1': {
        'level2': 'something',
    },
}
```
now if I would like to say that my key_claim_name is going to be `level1.level2` it says that's not possible through the following error:
```
{
    "message": "No mandatory 'level1.level2' in claims"
}

```

**Comment by jdartigalongue** [comment]
+1 I have the exact same issue.  
@meysam81 Did you ever find a solution to this?

**Comment by meysam81** [comment]
> +1 I have the exact same issue.
> @meysam81 Did you ever find a solution to this?

I ended up modifying the underlying source code in the docker container, committing it, and reusing it in my future deployments.

here's my modification:

In the docker container, open the following file:
`/usr/local/share/lua/5.1/kong/plugins/jwt/handler.lua`

replace the following code:
```lua
local jwt_secret_key = claims[conf.key_claim_name] or header[conf.key_claim_name]
```
with this one:
```lua
local jwt_secret_key = claims[conf.key_claim_name] or header[conf.key_claim_name] or (claims["CUSTOM_KEY"] and claims["CUSTOM_KEY"][conf.key_claim_name])
```
